### PR TITLE
Implement attribute to disable reports

### PR DESCRIPTION
**Pull request by Abyss777**
Lets start to implement features to hide some functionality in web-client.

Introduced attribute`web.disableReport` to hide report panel.

This is only feature I would make also respected by server. It will be additional protection for public use case. For example in taxi use case readonly user (customer) should not be able see history of car appointed to him. What do you think?

One comment in code...

### CODE REVIEW

File: web/app/controller/Root.js [hunk 1]
```diff
@@ -124,7 +124,15 @@ Ext.define('Traccar.controller.Root', {
         if (Traccar.app.isMobile()) {
             Ext.create('widget.mainMobile');
         } else {
-            Ext.create('widget.main');
+            main = Ext.create('widget.main');
+            if (Traccar.app.getAttributePreference('web.disableReport', false)) {
```
Comment (Abyss777) [inline web/app/controller/Root.js]: The best way is `main.lookupReference('reportView')` but unfortunately it does not work (return `null`) here is similar problem without solution https://www.sencha.com/forum/showthread.php?330708-Event-in-child-controller-damages-lookupReference-called-in-parent-controller

We can use `main.items.items[5]` directly but I'm afraid we can forget this and change layout.
Also this loop might be reused to hide Events panel if we implement it in future.

**Comment by Abyss777** [comment]
It is impossible to simply make `checkboxfield` return boolean value. Because of that unfortunately we always store boolean attributes as string. Added correctly check for both cases if value stored we as string or via API as boolean.

**Comment by Abyss777** [comment]
Sorry for asking. Have you had a chance to review this PR?

File: web/app/controller/Root.js [hunk 2]
```diff
@@ -124,7 +124,15 @@ Ext.define('Traccar.controller.Root', {
         if (Traccar.app.isMobile()) {
             Ext.create('widget.mainMobile');
         } else {
-            Ext.create('widget.main');
+            main = Ext.create('widget.main');
+            if (Traccar.app.getAttributePreference('web.disableReport', false).toString() === 'true') {
```
Comment (tananaev) [inline web/app/controller/Root.js]: I think this logic should be in the main layout itself.

File: web/app/store/ServerAttributes.js [hunk 3]
```diff
@@ -47,5 +47,9 @@ Ext.define('Traccar.store.ServerAttributes', {
         allowDecimals: false,
         minValue: Traccar.Style.mapDefaultZoom,
         maxValue: Traccar.Style.mapMaxZoom
+    }, {
+        key: 'web.disableReport',
```
Comment (tananaev) [inline web/app/store/ServerAttributes.js]: I think we should use more generic parameter naming. Something like `ui`.

**Comment by Abyss777** [comment]
You are right, logic in `Main` looks cleaner. Also renamed attribute.

File: web/app/view/Main.js [hunk 4]
```diff
@@ -27,6 +27,19 @@ Ext.define('Traccar.view.Main', {
         'Traccar.view.map.Map'
     ],
 
+    initComponent: function () {
+        var i;
+        if (Traccar.app.getAttributePreference('ui.disableReport', false).toString() === 'true') {
+            for (i = 0; i < this.items.length; i++) {
+                if (this.items[i].xtype === 'reportView') {
```
Comment (tananaev) [inline web/app/view/Main.js]: Can't we just get it by reference from here?

Comment (Abyss777) [inline web/app/view/Main.js]: Unfortunately not, `lookupReference` also return `null` from here

Comment (tananaev) [inline web/app/view/Main.js]: What about controller `init`? That's where we usually do things like this.

**Comment by tananaev** [comment]
Merged, thanks.

**Comment by Abyss777** [comment]
Sorry for did not check it myself. Lookup works.

**Comment by Abyss777** [comment]
What do you think about also respect this attribute on the server side?

**Comment by tananaev** [comment]
I think for now we just leave it on the client side. Don't want to add more complications on the server.

**Pull request merged**